core/templates/dev/head/domain/utilities/language-util.service.spec.ts

```diff
@@ -113,14 +113,22 @@ describe('Language util service', function() {
     });
 
   it('should correctly check if language supports autogenerated audio',
-    () => {
+    (done) => {
       // Chrome loads voices asynchronously.
       // https://stackoverflow.com/questions/21513706/
+      // Chrome also calls this multiple times and not all languages are
+      // populated the first time, see
+      // https://developer.mozilla.org/en-US/docs/Web/API/SpeechSynthesis
+      // for more details. In the case where all languages are loaded
+      // the first time, the test will complete in one pass successfully.
       window.speechSynthesis.onvoiceschanged = () => {
```

Comment:
Thanks! Two questions:

(a) did you confirm this test fails if the wrong code is used in browser-checker.service.ts?

(b) Is it possible to confirm that the onvoiceschanged() function was called? I can see this test passing (erroneously) if no execution of that function was made.

Reply:
(a) Yup, when I forgot to add a return https://github.com/oppia/oppia/pull/8233/files#diff-384e3f8384d830da8d028a89563df33fR31, the test failed on the `en` check. (Saying that false is not true)

(b) I tested to make sure that when I didn't call `done`, the test would fail (on a timeout).

Reply:
Ah sounds good. (b) is a bit subtle though. Can you add a comment explaining that the reason for the timeout is that, if done is not called, the test should fail due to timeout?